### DISCUSSION TITLE
Sideband is now more useful

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#78] [Show and tell]
Hello everyone
A lot has happened with the entire ecosystem over the last few weeks, and Sideband has also seen some love. With the latest release, it has gone from being primarily an experimental proof of concept app, to something that is actually more useful in a day to day setting. Don't get me wrong, it is still very much beta quality, and there is a lot of essentials missing that mobile-device users would normally expect, such as notifications and connection persistence in the background, but we are getting closer 👍 

You can now configure external connections directly in the Sideband app, so you can connect back to your Reticulum network over I2P or TCP over the Internet when on the go. It is now also much easier to install and run Sideband on desktop OSes with pip:
```
# Install
pip install sbapp

# And run
sideband
```

![Screenshot](https://github.com/markqvist/Sideband/raw/main/docs/screenshots/screenshot_1.png)

![Screenshot](https://github.com/markqvist/Sideband/raw/main/docs/screenshots/screenshot_3.png)

**Comment by githubques** [comment]
Installed Latest version (sbapp-0.1.8) on Ubuntu 22.04 and it doesn't load and instead gives this notice:

[INFO ] [Logger] Record Log in ......
[INFO ] [Kivy] v2.1.0
[INFO ] [Kivy] Installed at ...
[INFO ] [Python] v3.10.4 (main, date info) [GCC 11.2.0]
[INFO ] [Python] Interpreter at "user/bin/python3"
[INFO ] [Python] Purge log fired. Processing...
[INFO ] [Python] Purge finished!

Same happens when testing on 20.04. Only difference is python version:

[INFO ] [Python] v3.08.10 (default, date info)

**Reply by markqvist** [reply]
Thanks for the reports on failure to start up on Ubuntu. I have updated the wheel package, which was missing the correct reference to an internal module. You will have to uninstall Sideband, and reinstall from the repository (without using the local cached version). You can do this like so:

```
pip uninstall sbapp
pip install sbapp --no-cache-dir
```

That should make it work :)

**Reply by markqvist** [reply]
And make sure `pip` downloads the package named `sbapp-0.1.8-2-py3-none-any.whl`, which is the corrected build.

**Reply by githubques** [reply]
Worked! Nice setup.

Have installed on two cpus to test communication. Got warnings on one of them about conflicting scripts because of other things installed, but still works without issue.

**Reply by markqvist** [reply]
Good to hear! Thanks for reporting it back!

**Reply by githubques** [reply]
Has the .apk for SB been updated? Tried installing the original SB apk you posted a link to months back on an old Droid (no account/no sim) and it wouldn't allow it to load, but I was able to install a different apk without an issue.

**Reply by markqvist** [reply]
The Sideband APK should be version 0.1.8 and will run on Android 8.1 and up on ARM processors. It should install and run without any play store, google services or google API at all, and also without sim cards or any sort of registration. That is kinda the point :)

Really old Android devices might not work, because of the target Android API level of the build. I might try and support older versions of android at some point, but right now it is 8.1 minimum.

**Reply by githubques** [reply]
1) Thanks for the version number. I'll track down the old droid I have to see if it meets the criteria or not.

2) The annoying question for all the iphone users out there, but is there an ETA on when that version will be available?

3) Sideband is working great on Ubuntu 20.04 & 22.04. Took awhile to figure out SB has it's own address separate from Nomadnet found under Settings when you click the upper left hand menu. For some reason I thought SB was an extension of the original Nomadnet's LXMF address on the same device.

4) On Ubuntu 22.04 (but not 20.04) it gives an error message repeatedly while not interfering with performance:

[DATE-TIME] [Notice] Reticulum started, activating LXMF...
[DATE-TIME] [Notice] Active propagation node set to: <[letters&numbers]>
[DATE-TIME] [Notice] Saving Sideband configuration...
[DATE-TIME] [Notice] Sideband Core <sbapp.sideband.core.SidebandCore object at [numbers&letters]> started
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
WARNING: running xinput against an Xwayland server. See the xinput man page for details.

Like I said, it doesn't seem to hurt performance, but I thought you might want to know in case it becomes an issue for others.
